gorm/transaction_test.go

```diff
@@ -45,6 +53,95 @@ func TestUnaryServerInterceptor_success(t *testing.T) {
 	}
 }
 
+func TestUnaryServerInterceptor_with_readonlydb(t *testing.T) {
```

Comment:
Actually wondering if we need this test and the next one. They don't call `BeginFromContext()` where all the changes have been made.